### PR TITLE
Add Queries to access usage of fetchpriority='high' on LCP image

### LINKED ISSUE #5: Query to assess usage of `fetchpriority="high"` on LCP image
* Create HTTP Archive query to get **% of WordPress sites that use `fetchpriority="high"` on LCP image**
* Also create more general query to assess **% of WordPress sites that even have an image as the LCP element**

### PR DESCRIPTION
**Pull request by kt-12**
Fixes #5 

> **Note:** *At the time of running this query, the latest httparchive reports available were of `01, November 2022`, which were used in all the queries here*.
----
 

> **Overall Opportunity Score %** = *100 x ( No. of WordPress sites having LCP image  **without** fetchpriority='high') / (**Total no. of** WordPress sites)*
### Result
client | with_fetchpriority_on_lcp | without_fetchpriority_on_lcp | total_with_lcp | total_wp_sites | opportunity | overall_opportunity
-- | -- | -- | -- | -- | -- | --
desktop | 444 | 1818969 | 1819413 | 4245886 | 99.976 % | 42.841 %
mobile | 628 | 2189295 | 2189923 | 5707622 | 99.971 % | 38.357 %

### CODE REVIEW

**Comment by felixarntz** [comment]
@kt-12 Would it be possible to enhance the queries to also come with an additional column for the % of total WP sites (similar to #13 for example)? That would be neat.

Alternatively we could just query those numbers manually of course and calculate the results ourselves. But doing it in the query and having the results at a glance would be extra points. :)

File: sql/2022/12/count-sites-having-fetchpriority-high-on-lcp-img.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/15
+
+SELECT 
+  lh._TABLE_SUFFIX as client, 
+  count(lh.url) as sites 
+FROM 
+  `httparchive.lighthouse.2022_10_01_*` as lh 
+  JOIN `httparchive.technologies.2022_10_01_*` as tech ON tech.url = lh.url 
+where
```
Comment (felixarntz) [inline sql/2022/12/count-sites-having-fetchpriority-high-on-lcp-img.sql:25]: Let's use capitalization consistently, having such keywords in all uppercase. That also applies to a few other instances below and in the other query.

File: sql/2022/12/count-sites-having-fetchpriority-high-on-lcp-img.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/15
+
+SELECT 
+  lh._TABLE_SUFFIX as client, 
+  count(lh.url) as sites 
+FROM 
+  `httparchive.lighthouse.2022_10_01_*` as lh 
+  JOIN `httparchive.technologies.2022_10_01_*` as tech ON tech.url = lh.url
```
Comment (felixarntz) [inline sql/2022/12/count-sites-having-fetchpriority-high-on-lcp-img.sql:24]: Let's keep this indented in a way that is easier to read, with the main keyword at the start of a new line.
```suggestion
JOIN
  `httparchive.technologies.2022_10_01_*` as tech
ON
  tech.url = lh.url 
```

File: sql/2022/12/count-sites-having-fetchpriority-high-on-lcp-img.sql [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  and category = 'CMS' 
+  and REGEXP_CONTAINS(
+    JSON_EXTRACT(
+      report, '$.audits.largest-contentful-paint-element.details.items'
+    ), 
+    '<img.*fetchpriority.{3}high'
+  ) 
+group by 
+  lh._TABLE_SUFFIX;
+
```
Comment (felixarntz) [inline sql/2022/12/count-sites-having-fetchpriority-high-on-lcp-img.sql:37]: No need for a blank line here.
```suggestion
```

**Review by felixarntz — COMMENTED** [review]
@kt-12 A few nit-pick comments on the query. Let me know what you think about my other more general feedback above in regards to % of total.

**Comment by kt-12** [comment]
@felixarntz  I have come up with one single query to calculate Opportunity Score and Overall Opportunity score

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql [hunk 4]
```diff
@@ -0,0 +1,96 @@
+# HTTP Archive query to get Performance Lab plugin version distribution in a given month.
```
Comment (mukeshpanchal27) [inline sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:1]: @kt-12 the comment needs to update. Something like below. Please update it as per your result.

```suggestion
# HTTP Archive query to get usage of LCP image measurements in a given month.
```

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  JOIN 
+    `httparchive.technologies.2022_10_01_*` AS tech 
+  ON 
+    tech.url = lh.url 
+  WHERE 
+    lh._TABLE_SUFFIX = tech._TABLE_SUFFIX 
+  AND 
+    app = 'WordPress' 
+  AND 
+    category = 'CMS'
```
Comment (mukeshpanchal27) [inline sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:48]: Is it necessary to pass category if we get a proper result by using the "app" field?

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    ), 
+    '<img.*fetchpriority.{3}high'
+  ) 
+GROUP BY 
+  lh._TABLE_SUFFIX,
+  total,
+  totalwp
+ORDER BY 
+  Client
+ASC
```
Comment (mukeshpanchal27) [inline sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:96]:
```suggestion
  Client ASC

```
Add an empty line at end.

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql (hunk 5 above)
Comment (kt-12) [reply sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:48]: @mukeshpanchal27  there here 2 categories for WordPress for each URL. So, if we only pass app, it will return us two rows for a single URL in the results by default (unless you are using distinct). Explicit mentioning of category gets us only one row and thereby the entire query runs a bit faster.

**Review by felixarntz — COMMENTED** [review]
@kt-12 Query and PR overall looks good to me. However there is still the open question: Should we use this query, or should we use https://github.com/GoogleChromeLabs/wpp-research/issues/5#issuecomment-1359419126?

Additionally, you will need to sign the CLA in order for this to be merged. Please coordinate that with @eclarke1 who should be able to help with that.

**Comment by kt-12** [comment]
@felixarntz  I am coming with the new query today, keeping [#5 comment](https://github.com/GoogleChromeLabs/wpp-research/issues/5#issuecomment-1359419126?) as the base.

**Comment by kt-12** [comment]
@felixarntz  I have updated the query. The score is near about the same.  However, it was observed that the LCP elements detected by Lighthouse and WebPageTest differed for some sites.  WebPageTest report being more structured and detailed for the LCP element, was my final choice for this query.

The new scores are updated in the table above.

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# HTTP Archive query to get % of WordPress sites not having fetchpriority='high' on LCP image.
```
Comment (felixarntz) [inline sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:1]: Can you rename this file to a more concise name? For example `lcp-image-without-fetchpriority-high-opportunity.sql`.

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+}
+''';
+
+SELECT
+  client,
+  with_fetchpriority_on_lcp,
+  (total_with_lcp-with_fetchpriority_on_lcp) AS without_fetchpriority_on_lcp,
+  total_with_lcp,
+  total_wp_sites,
+  CONCAT(ROUND((total_with_lcp-with_fetchpriority_on_lcp)*100/total_with_lcp, 3),' %') AS opportunity_score,
```
Comment (felixarntz) [inline sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:34]: Can we shorten this to just `opportunity`? It's not really a score, just "opportunity" is sufficient.

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+  CONCAT(ROUND((total_with_lcp-with_fetchpriority_on_lcp)*100/total_wp_sites, 3),' %') AS overall_opportunity_score
```
Comment (felixarntz) [inline sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql:35]: Same here, this can become just `overall_opportunity`.

**Review by felixarntz — CHANGES_REQUESTED** [review]
@kt-12 The query looks solid. However, something appears to be wrong with the PR as it apparently re-includes a ton of other unrelated changes. Can you fix it so that it just has the new query file and the addition to `sql/README.md`? Make sure to also merge in latest `main` to resolve the conflict.

Other than that, just a few small comments below on making some names more concise.

File: sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql (hunk 7 above)
Comment (kt-12) [reply sql/2022/12/opportunity-score-wp-sites-lcp-without-fetchpriority-high.sql]: Changed

**Review by felixarntz — DISMISSED** [review]
Great, thanks @kt-12!

**Comment by felixarntz** [comment]
@mukeshpanchal27 @10upsimon @jjgrainger Can one of you please give this a 2nd review and approval? :)

**Review by jjgrainger — DISMISSED** [review]
Looks good, thanks @kt-12

**Review by 10upsimon — DISMISSED** [review]

**Review by mukeshpanchal27 — DISMISSED** [review]
Thanks @kt-12, LGTM!

Not blocker. The MORE menu item in the BigQuery editor contains a query format option; it's a good option to use to format the query and it also fixes the indentation issue.

cc. @felixarntz

**Comment by kt-12** [comment]
> Thanks @kt-12, LGTM!
> 
> Not blocker. The MORE menu item in the BigQuery editor contains a query format option; it's a good option to use to format the query and it also fixes the indentation issue.
> 
> cc. @felixarntz

Thanks, @mukeshpanchal27  for pointing that out. I agree, this way we can also have a standard in place for all the BigQuery. I have made the change and have pushed it. 
cc: @felixarntz

**Review by felixarntz — APPROVED** [review]
Thanks @kt-12!

**Review by 10upsimon — APPROVED** [review]
